lib/asciidoctor/substitutors.rb

```diff
@@ -948,7 +948,7 @@ def highlight_source source, process_callouts
     if (linenums_mode = (attr? 'linenums') ? (doc_attrs[%(#{syntax_hl_name}-linenums-mode)] || :table).to_sym : nil)
       start_line_number = 1 if (start_line_number = (attr 'start', 1).to_i) < 1
     end
-    highlight_lines = resolve_lines_to_highlight source, (attr 'highlight') if attr? 'highlight'
+    highlight_lines = resolve_lines_to_highlight source, (attr 'highlight'), start_line_number if attr? 'highlight'
```

Comment:
I'm trying to figure out if it makes more sense to pass in the start value, or to remove it from each value in the result. I suppose it's elegant to pass it in.

Reply:
`start_line_number` can be `nil`, should we handle this case in `resolve_lines_to_highlight`?

```rb
start && start > 1 ? (lines.sort.map {|it| it - start + 1 }) : lines.sort
```

As an alternative, we could state in the documentation that `start` must be a positive integer greater or equals to 1?

Reply:
Good point. In that case, the method argument should be `start = nil` and the logic should be:

```ruby
unless (shift = start ? start - 1 : 0) == 0
  lines = lines.map {|it| it - shift }
end
lines.sort
```